io.openems.edge.core/src/io/openems/edge/core/timer/TimerManager.java

```diff
@@ -0,0 +1,185 @@
+    }
+
+    /**
+     * Get a timer to count every call to the timer.check() or the
+     * timer.checkAndRestart() method.
+     * 
+     * @param channel         if this channel is provided, this channel is used to
+     *                        hold the state of the timer.
+     * @param countCheckCalls the count to use.
+     * @return true if the given count has been reached, false else
+     */
+    public default Timer getTimerByCount(Channel<Integer> channel, int countCheckCalls) {
+	return this.getTimerByCount(channel, countCheckCalls, 0);
+    }
+
+    /**
+     * Get a timer to count every call to the timer.check() or the
+     * timer.checkAndRestart() method.
+     * 
+     * @param channel          if this channel is provided, this channel is used to
+     *                         hold the state of the timer.
+     * @param countCheckCalls  the count to use.
+     * @param startDelayInSecs the timer will return false, as long as the
+     *                         startDelayInSecs is not reached. This is especially
+     *                         useful for heaters which are very sensitive against
+     *                         frequent condition changes (can happen in systems
+     *                         which restart OpenEMS frequently).
+     * @return true if the given count has been reached, false else
+     */
+    public Timer getTimerByCount(Channel<Integer> channel, int countCheckCalls, int startDelayInSecs);
+
+    /**
+     * Get a timer to count every core cycle.
+     * 
+     * @param count the count to use.
+     * 
+     * @return true if the given count has been reached, false else
+     */
+    public default Timer getTimerByCoreCycles(int count) {
+	return this.getTimerByCoreCycles(null, count, 0);
+    }
+
+    /**
+     * Get a timer to count every core cycle.
+     * 
+     * @param count            the count to use.
+     * @param startDelayInSecs the timer will return false, as long as the
+     *                         startDelayInSecs is not reached. This is especially
+     *                         useful for heaters which are very sensitive against
+     *                         frequent condition changes (can happen in systems
+     *                         which restart OpenEMS frequently).
+     * @return true if the given count has been reached, false else
+     */
+    public default Timer getTimerByCoreCycles(int count, int startDelayInSecs) {
+	return this.getTimerByCoreCycles(null, count, startDelayInSecs);
+    }
+
+    /**
+     * Get a timer to count every core cycle.
+     * 
+     * @param channel if this channel is provided, this channel is used to hold the
+     *                state of the timer.
+     * @param count   the count to use.
+     * @return true if the given count has been reached, false else
+     */
+    public default Timer getTimerByCoreCycles(Channel<Integer> channel, int count) {
+	return this.getTimerByCoreCycles(channel, count);
```

Comment:
Shouldn't this be `getTimerByCoreCycles(channel, count, 0)`? Otherwise the method calls itself.